src/NServiceBus.Core/Sagas/SagaMessage.cs

```diff
@@ -10,16 +12,22 @@ public class SagaMessage
         /// </summary>
         /// <param name="messageType">Type of the message.</param>
         /// <param name="isAllowedToStart"><code>true</code> if the message can start the saga, <code>false</code> otherwise.</param>
-        public SagaMessage(string messageType, bool isAllowedToStart)
+        public SagaMessage(Type messageType, bool isAllowedToStart)
         {
             MessageType = messageType;
+            MessageTypeName = messageType.FullName;
             IsAllowedToStartSaga = isAllowedToStart;
         }
 
         /// <summary>
         /// The type of the message.
         /// </summary>
-        public string MessageType { get; private set; }
+        public Type MessageType { get; private set; }
```

Comment:
Having the type allows me to compare on type, obviously, but then also gives me the ability to use the simple type name in exception messages to make them really useful and prescriptive.